### ISSUE TITLE
Set up Ubuntu 24.04 systems to be able to support it for Temurin

### ISSUE THREAD

**Issue by sxa** [adoptium/infrastructure#3501]
Ubuntu 24.04 will be out later this month. We should look at our existing inventory and make some of these available. This will involve:
- [x] Identifying older systems that can be replaced with Ubuntu 24.04 without restricting testing.
- [x] Having at least one static docker image on each architecture
- [x] Ideally having one real machine for each architecture on this
- [x] Verifying that it is suitable as a base for our dockerhost system and deploying them: https://github.com/adoptium/infrastructure/issues/3585
- [ ] [Add Ubuntu 24.04 to Vagrant Playbook Check job](https://github.com/adoptium/infrastructure/issues/3577)

**Comment by Haroon-Khel** [comment]
Ubuntu 2404 static docker container online in jenkins https://ci.adoptium.net/computer/test-docker-ubuntu2404-x64-1/

Will run the aqa test pipeline on it shortly

**Comment by Haroon-Khel** [comment]
https://ci.adoptium.net/computer/test-docker-ubuntu2404-armv8-1/ https://ci.adoptium.net/job/AQA_Test_Pipeline/239/console
https://ci.adoptium.net/computer/test-docker-ubuntu2404-s390x-1/ https://ci.adoptium.net/job/AQA_Test_Pipeline/240/console

Getting unusual errors while building the ppc64le and arm32 images

```
 > [ 7/25] RUN mkdir -p /usr/lib/jvm/jdk17 && tar -xpzf /tmp/jdk17.tar.gz -C /usr/lib/jvm/jdk17 --strip-components=1:                                                                                               
0.295 tar: conf/security/policy/unlimited: Cannot change mode to rwxr-xr-x: Operation not permitted                                                                                                                 
0.295 tar: conf/security/policy/limited: Cannot change mode to rwxr-xr-x: Operation not permitted                                                                                                                   
0.295 tar: conf/security/policy: Cannot change mode to rwxr-xr-x: Operation not permitted                                                                                                                           
0.295 tar: conf/security: Cannot change mode to rwxr-xr-x: Operation not permitted                                                                                                                                  
0.295 tar: conf/sdp: Cannot change mode to rwxr-xr-x: Operation not permitted
0.296 tar: conf/management: Cannot change mode to rwxr-xr-x: Operation not permitted
0.296 tar: conf: Cannot change mode to rwxr-xr-x: Operation not permitted
0.305 tar: legal/java.base: Cannot change mode to rwxr-xr-x: Operation not permitted
1.052 tar: jmods: Cannot change mode to rwxr-xr-x: Operation not permitted
```

**Comment by Haroon-Khel** [comment]
Got the remaining nodes up and running by building them manually

https://ci.adoptium.net/computer/test-docker-ubuntu2404-ppc64le-1/
https://ci.adoptium.net/job/AQA_Test_Pipeline/242/console

https://ci.adoptium.net/computer/test-docker-ubuntu2404-armv7l-1/
https://ci.adoptium.net/job/AQA_Test_Pipeline/241/console

**Comment by sxa** [comment]
Unfortunately we don't have an easy way of provisioning a 24.04 on arm32 or ppc64le at the moment to identify whether this is specific to running in a container.

**Comment by sxa** [comment]
It looks like we're getting some interesting new errors in the [sanity.openjdk](https://ci.adoptium.net/job/Test_openjdk17_hs_sanity.openjdk_x86-64_linux/433/testReport/)
Also in the [extended run](https://ci.adoptium.net/job/Test_openjdk17_hs_extended.openjdk_x86-64_linux/181/consoleFull) we're getting quite a lot of failures some of which are related to `dpkg` so are likely related to the new version (I'm quite surprised that the tests are doing things at that OS-specific level, but apparently they do!):
```
 [09:14:04.910] Running dpkg
 [09:14:06.077] Command [PID: 1034473]:
     dpkg -S /lib/x86_64-linux-gnu/libbrotlidec.so.1
 [09:14:06.077] Output:
     dpkg-query: no path found matching pattern /lib/x86_64-linux-gnu/libbrotlidec.so.1
 [09:14:06.077] Returned: 1
 
 [09:14:06.078] java.io.IOException: Command [dpkg, -S, /lib/x86_64-linux-gnu/libbrotlidec.so.1] exited with 1 code
```
Based on this I think it's worth getting a "real" ubuntu 24.04 provisioned somewhere and verify that nothing we're seeing is the result of using an earlier kernel version with a newer container.

**Comment by Haroon-Khel** [comment]
OSUOSL's openstack and azure dont seem to have a ubuntu2404 image (yet)

**Comment by sxa** [comment]
This is where having the ESXi server was useful :-)

**Comment by sxa** [comment]
>  Will run the aqa test pipeline on it shortly
> EDIT: https://ci.adoptium.net/job/AQA_Test_Pipeline/237/console

To keep the machines busy over teh UK bank holiday weekend I've kicked that off again with all the other versions at https://ci.adoptium.net/job/AQA_Test_Pipeline/243/

**Comment by sxa** [comment]
> Got the remaining nodes up and running by building them manually
> 
> https://ci.adoptium.net/job/AQA_Test_Pipeline/242/console (ppc64le)
> https://ci.adoptium.net/job/AQA_Test_Pipeline/241/console (arm32)

Noting that those seem to be failing after the tarball extraction had the same problem you had with the docker image creation

**Comment by Haroon-Khel** [comment]
Ive marked both machines offline so they dont affect the weekend nightlies

**Comment by sxa** [comment]
I've just ran the Dockerfile for s390x and ppc64le in an emulated docker on an arm64 Ubuntu 24.04 host and they both worked without any permission issues 🤔

**Comment by sxa** [comment]
I'm doing some re-runs based on the "clean" (green) sanity.openjdk jobs that I was doing as part of https://github.com/adoptium/temurin-build/issues/3685#issuecomment-2075133310 (the x64 -cfi-vh row) with JDK22+35:
- [Same machine as before](https://ci.adoptium.net/job/Grinder/9919/) ([test-docker-ubuntu2004-x64-2](https://ci.adoptium.net/computer/test-docker-ubuntu2004-x64-2)
- [The new Ubuntu 24.04 system](https://ci.adoptium.net/job/Grinder/9920/)
- [The new Amazon Linux 2023 system](https://ci.adoptium.net/job/Grinder/9921/) (for comparison)

**Comment by sxa** [comment]
> Ubuntu 2404 static docker container online in jenkins https://ci.adoptium.net/computer/test-docker-ubuntu2404-x64-1/
> 
> Will run the aqa test pipeline on it shortly
> 
> EDIT: https://ci.adoptium.net/job/AQA_Test_Pipeline/237/console

> https://ci.adoptium.net/computer/test-docker-ubuntu2404-armv8-1/ https://ci.adoptium.net/job/AQA_Test_Pipeline/239/console
> https://ci.adoptium.net/computer/test-docker-ubuntu2404-s390x-1/ https://ci.adoptium.net/job/AQA_Test_Pipeline/240/console

On the basis of the reruns in the previous comment running clean I'm going to re-run these (including extra versions) with `ADOPTOPENJDK_BRANCH=v1.0.1-release` so that the material matches the product under test (`SDK_RESOURCE=releases`). These are being run with all versions: 17,8,22,21,11. The original JDK17 runs linked above took 29 hours (x64), 14h (aarch64) and 17 hours (s390x). so the first 2-3 versions will hopefully be mostly complete on Monday morning which will hopefully give enough of a result set that we can determine if there are likely to be any problems.

- [x64 - 254](https://ci.adoptium.net/job/AQA_Test_Pipeline/254/)
- [aarch64 - 257](https://ci.adoptium.net/job/AQA_Test_Pipeline/257/)
- [s390x - 256](https://ci.adoptium.net/job/AQA_Test_Pipeline/256/)

**Comment by sxa** [comment]
> OSUOSL's openstack and azure dont seem to have a ubuntu2404 image (yet)

We should look at provisioning a 22.04 system at OSUOSL (Maybe for Aarch64 and ppc64le both at OSUOSL) and then upgrading it to 24.04 within the OS and see if that makes a difference

**Comment by sxa** [comment]
Deep dive into the `tools/jpackage` failures which are occurring on 24.04 but not on 22.04:

On both machines `dpkg -S /lib/aarch64-linux-gnu/libmd.so.0` are failing, although on the 22.04 machine it gives a summary of `[14:00:33.690] Required packages: [libc6, zlib1g]` but on 24.04 it has `[14:00:38.972] Required packages: []

This is not considered a failure, but later on when it tries to build it's own `appcategorytest` package it fails on:

- `ERROR: Failed: Check [libc] package is in the list of required packages [] of [appcategorytest] package`
where in the passing case it has:
- `TRACE: assertTrue(): Check [libc] package is in the list of required packages [libc6, zlib1g] of [appcategorytest] package`

OK Further investigation - all but five of the `dpkg` commands run on the Ubuntu 22.04 machine also give a non-zero return code. The ones that succeed are:
- dpkg -S /lib/aarch64-linux-gnu/libc.so.0
- dpkg -S /lib/aarch64-linux-gnu/libm.so.6
- dpkg -S /lib/aarch64-linux-gnu/libz.so.1
- dpkg -S /lib/aarch64-linux-gnu/librt.so.1
- dpkg -S /lib/aarch64-linux-gnu/libdl.so.2
All of which apart from `libz.so.1` are in `libc6`, and indeed if you run `dpkg -L libc6` you don't get the `/lib` versions on Ubuntu 24.04. Simple reproduce `docker run ubuntu:24.04 dpkg -L libc6`

**Comment by jiekang** [comment]
I've seen fixes to the jpackage tests that reference specific quirks with new versions of platforms like Fedora. Is this failure similar? Can we fix the test upstream?

**Comment by sxa** [comment]
> I've seen fixes to the jpackage tests that reference specific quirks with new versions of platforms like Fedora. Is this failure similar? Can we fix the test upstream?

I would expect so, yes.

**Comment by sxa** [comment]
Something like `dpkg -S <blah> || dpkg -S $(realpath -S <blah>)` is possibly enough to work around this problem - `realpath` seems to be in the base Ubuntu images so it likely safe, although this code may be within the jpackage tool and so it may not be as simple as adding a construct like that into the test case.

**Comment by jiekang** [comment]
Okay, well at the least we can report this in a JBS bug upstream.

**Comment by Haroon-Khel** [comment]
> OSUOSL's openstack and azure dont seem to have a ubuntu2404 image (yet)

Azure and OSUOSL now have ubuntu2404 images, so we can have actual vm (non static docker node) ubuntu 2404 machines up

**Comment by Haroon-Khel** [comment]
**JDK21 test failures**

**sanity openjdk**

jdk_lang

```
TEST: java/lang/ProcessBuilder/JspawnhelperWarnings.java
TEST: java/lang/Thread/virtual/stress/GetStackTraceALotWhenPinned.java#id0
TEST: java/lang/Thread/virtual/JfrEvents.java
TEST: java/lang/Thread/virtual/TracePinnedThreads.java
TEST: java/lang/Thread/virtual/ThreadAPI.java#default
```

jdk_security1
```
TEST: java/security/Security/ConfigFileTest.java
```
jdk_util

TEST: jdk/internal/util/ReferencedKeyTest.java

jdk_foreign
```
TEST: java/foreign/TestLargeSegmentCopy.java
```
**extended openjdk**

hotspot_serviceability_jvmti

```
TEST: serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java#default
TEST: serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java#no-vmcontinuations
TEST: serviceability/jvmti/GetOwnedMonitorInfo/GetOwnedMonitorInfoTest.java
TEST: serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java
```

jvm_compiler

```
TEST: compiler/loopopts/TestRemixAddressExpressionsWithIrreducibleLoop.java
TEST: compiler/types/TestSubTypeCheckWithBottomArray.java#stress
TEST: compiler/types/TestSubTypeCheckWithBottomArray.java#Xbatch
TEST: compiler/types/TestSubTypeCheckWithBottomArray.java#Xcomp
```

jdk_other
```
TEST: com/sun/jndi/ldap/LdapSSLHandshakeFailureTest.java
```
jdk_net

```
TEST: java/net/httpclient/whitebox/SSLFlowDelegateTestDriver.java
TEST: java/net/httpclient/HttpInputStreamAvailableTest.java
```

jdk_security3

```
TEST: javax/net/ssl/TLSv13/EngineOutOfSeqCCS.java
TEST: jdk/security/logging/RecursiveEventHelper.java
TEST: sun/security/ssl/SSLEngineImpl/TestBadDNForPeerCA.java
TEST: sun/security/ssl/SSLEngineImpl/TestBadDNForPeerCA12.java
```

jdk_tools

```
TEST: tools/jpackage/linux/AppAboutUrlTest.java#id0
TEST: tools/jpackage/linux/AppCategoryTest.java
TEST: tools/jpackage/linux/LinuxBundleNameTest.java
TEST: tools/jpackage/linux/LinuxResourceTest.java
TEST: tools/jpackage/linux/jdk/jpackage/tests/UsrTreeTest.java
TEST: tools/jpackage/linux/MaintainerTest.java
TEST: tools/jpackage/linux/PackageDepsTest.java
TEST: tools/jpackage/linux/ReleaseTest.java#id0
TEST: tools/jpackage/linux/ShortcutHintTest.java#id0
TEST: tools/jpackage/share/AddLShortcutTest.java
TEST: tools/jpackage/share/AddLauncherTest.java#id1
TEST: tools/jpackage/share/AppImagePackageTest.java
TEST: tools/jpackage/share/AppContentTest.java
TEST: tools/jpackage/share/EmptyFolderPackageTest.java
TEST: tools/jpackage/share/LicenseTest.java#id1
TEST: tools/jpackage/share/FileAssociationsTest.java#id0
TEST: tools/jpackage/share/PerUserCfgTest.java
TEST: tools/jpackage/share/ServiceTest.java
TEST: tools/launcher/RunpathTest.java
TEST: tools/launcher/Settings.java
```

jdk_jfr
```
TEST: jdk/jfr/jcmd/TestJcmdDump.java
```
jdk_jdi

```
TEST: com/sun/jdi/FinalizerTest.java
```

**JDK 17 test failures**

**sanity openjdk**

jdk_lang

```
TEST: java/lang/invoke/MethodHandleProxies/Driver.java
```

jdk_security1

```
TEST: java/security/Security/ConfigFileTest.java
```

jdk_util

```
TEST: java/util/Locale/LanguageSubtagRegistryTest.java
TEST: java/util/Locale/LSRDataTest.java
```

**extended openjdk**

jdk_other

```
TEST: com/sun/jndi/ldap/LdapSSLHandshakeFailureTest.java
```

jdk_net

```
TEST: java/net/httpclient/HttpInputStreamAvailableTest.java
```

jdk_nio

```
TEST: java/nio/file/spi/SetDefaultProvider.java
```

jdk_security3

```
TEST: javax/net/ssl/DTLS/DTLSWontNegotiateV10.java
TEST: sun/security/ssl/SSLContextImpl/SSLContextDefault.java
TEST: jdk/security/logging/RecursiveEventHelper.java
```

jdk_tools

```
TEST: tools/jlink/plugins/CDSPluginTest.java
TEST: tools/jpackage/linux/AppAboutUrlTest.java#id0
TEST: tools/jpackage/linux/AppCategoryTest.java
TEST: tools/jpackage/linux/LinuxBundleNameTest.java
TEST: tools/jpackage/linux/LinuxResourceTest.java
TEST: tools/jpackage/linux/LinuxWeirdOutputDirTest.java
TEST: tools/jpackage/linux/jdk/jpackage/tests/UsrTreeTest.java
TEST: tools/jpackage/linux/MaintainerTest.java
TEST: tools/jpackage/linux/PackageDepsTest.java
TEST: tools/jpackage/linux/ReleaseTest.java
TEST: tools/jpackage/linux/ShortcutHintTest.java#id0
TEST: tools/jpackage/share/AppImagePackageTest.java
TEST: tools/jpackage/share/AddLauncherTest.java#id1
TEST: tools/jpackage/share/EmptyFolderPackageTest.java
TEST: tools/jpackage/share/LicenseTest.java#id1
TEST: tools/jpackage/share/FileAssociationsTest.java#id0
TEST: tools/launcher/Settings.java
```

jdk_jfr

```
TEST: jdk/jfr/event/gc/detailed/TestGCCPUTimeEvent.java#Parallel
TEST: jdk/jfr/event/gc/detailed/TestGCCPUTimeEvent.java#G1
TEST: jdk/jfr/event/gc/detailed/TestGCCPUTimeEvent.java#Serial
TEST: jdk/jfr/event/metadata/TestLookForUntestedEvents.java
TEST: jdk/jfr/event/runtime/TestActiveSettingEvent.java
```

**Comment by Haroon-Khel** [comment]
Set up a new x64 ubuntu2404 machine in azure https://ci.adoptium.net/computer/test-azure-ubuntu2404-x64-1/

https://ci.adoptium.net/job/AQA_Test_Pipeline/266/console

**Comment by Haroon-Khel** [comment]
test-azure-ubuntu2404-x64-1 jdk21 test failures

**extended openjdk**

```
hotspot_serviceability_jvmti:
TEST: serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java#default
TEST: serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java#no-vmcontinuations
TEST: serviceability/jvmti/GetOwnedMonitorInfo/GetOwnedMonitorInfoTest.java
TEST: serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java
```

```
jvm_compiler
TEST: compiler/loopopts/superword/TestLargeScaleAndStride.java#AlignVector
TEST: compiler/loopopts/TestRemixAddressExpressionsWithIrreducibleLoop.java
TEST: compiler/rangechecks/TestLargeScaleInLongRCOverflow.java
TEST: compiler/types/TestSubTypeCheckWithBottomArray.java#stress
TEST: compiler/types/TestSubTypeCheckWithBottomArray.java#Xbatch
TEST: compiler/types/TestSubTypeCheckWithBottomArray.java#Xcomp
```

```
jdk_other
com/sun/jndi/ldap/LdapSSLHandshakeFailureTest.java
```

```
jdk_net
TEST: java/net/httpclient/whitebox/SSLFlowDelegateTestDriver.java
TEST: java/net/httpclient/HttpInputStreamAvailableTest.java
```

```
jdk_security3
TEST: javax/net/ssl/TLSv13/EngineOutOfSeqCCS.java
TEST: sun/security/ssl/SSLEngineImpl/TestBadDNForPeerCA.java
TEST: sun/security/ssl/SSLEngineImpl/TestBadDNForPeerCA12.java
TEST: jdk/security/logging/RecursiveEventHelper.java
```

29 jdk_tools failures

```
jdk_jfr
jdk/jfr/jcmd/TestJcmdDump.java
```

```
jdk_jdi
com/sun/jdi/FinalizerTest.java
```

**sanity openjdk**

```
jdk_lang
TEST: java/lang/ProcessBuilder/JspawnhelperWarnings.java
TEST: java/lang/String/CompactString/NegativeSize.java
TEST: java/lang/Thread/virtual/stress/GetStackTraceALotWhenPinned.java#id0
TEST: java/lang/Thread/virtual/JfrEvents.java
TEST: java/lang/Thread/virtual/TracePinnedThreads.java
TEST: java/lang/Thread/virtual/ThreadAPI.java#default
```

```
jdk_security1
java/security/Security/ConfigFileTest.java
```

```
jdk_util
jdk/internal/util/ReferencedKeyTest.java
```

**Comment by sxa** [comment]
Ref the last comment it would be good to have a similar summary for the docker container run mentioned in https://github.com/adoptium/infrastructure/issues/3501#issuecomment-2090349452 before the jobs for those disappear :-)

Additional note, I've split out another issue - linked in the description - for covering the creation/update of dockerhost Ubuntu 24.04 systems so that this issue can be focused on getting Ubuntu 24.04 into a sufficiently good state that we can be confident that we can [start supporting it as an official docker image](https://github.com/adoptium/containers/issues/532).

**Comment by Haroon-Khel** [comment]
test-docker-ubuntu2404-x64-1 jdk 17 failures

**sanity openjdk**

```
jdk_lang
 java/lang/String/CompactString/NegativeSize.java.NegativeSize
 java/lang/invoke/MethodHandleProxies/Driver.java.Driver
```

```
jdk_security1
 java/security/Security/ConfigFileTest.java.ConfigFileTest
```

```
jdk_util
 java/util/Locale/LSRDataTest.java.LSRDataTest
 java/util/Locale/LanguageSubtagRegistryTest.java.LanguageSubtagRegistryTest

```
**extended openjdk**

```
jdk_net
 java/net/httpclient/HttpInputStreamAvailableTest.java.HttpInputStreamAvailableTest
 javax/net/ssl/DTLS/DTLSWontNegotiateV10.java.DTLSWontNegotiateV10
```

```
jdk_nio
java/nio/file/spi/SetDefaultProvider.java.SetDefaultProvider
```

```
jdk_security3
 jdk/security/logging/RecursiveEventHelper.java.RecursiveEventHelper
 sun/security/ssl/SSLContextImpl/SSLContextDefault.java.SSLContextDefault
```

jdk_tools 26 failures

```
jdk_jfr
 jdk/jfr/event/gc/detailed/TestGCCPUTimeEvent.java#G1.TestGCCPUTimeEvent_G1
 jdk/jfr/event/gc/detailed/TestGCCPUTimeEvent.java#Parallel.TestGCCPUTimeEvent_Parallel
 jdk/jfr/event/gc/detailed/TestGCCPUTimeEvent.java#Serial.TestGCCPUTimeEvent_Serial
 jdk/jfr/event/metadata/TestLookForUntestedEvents.java.TestLookForUntestedEvents
 jdk/jfr/event/runtime/TestActiveSettingEvent.java.TestActiveSettingEvent
```

**Comment by Haroon-Khel** [comment]
JDK17 aqa pipeline using v1.0.1-release branch

   - test-azure-ubuntu2404-x64-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/268/consolet
    - test-docker-ubuntu2404-x64-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/269/console
    - test-docker-ubuntu2404-armv7-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/270/console
    - test-docker-ubuntu2404-armv8-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/272/console
    - test-docker-ubuntu2404-s390x-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/273/console
    - test-docker-ubuntu2404-armv7l-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/274/console
    - test-docker-ubuntu2404-ppc64le-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/275/console
        - Still seeing unzip, untar errors
    - test-osuosl-ubuntu2404-aarch64-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/276/console

**Comment by sxa** [comment]
Raised issue referenced above for the `jpackage` failures which are definitely a result of a change on the Ubuntu side rendering the openjdk tests incompatible with 24.04 for now.

**Comment by sxa** [comment]
> JDK17 aqa pipeline using v1.0.1-release branch
> 
> test-azure-ubuntu2404-x64-1 - https://ci.adoptium.net/job/AQA_Test_Pipeline/268/consolet test-docker-ubuntu2404-x64-1

Since the machine is now idle can we get some other pipelines kicked off with other versions. Since the goal here is to have this as an official docker image for all JDK versions we need to have confidence that we know if there are any issues being able to declare it as good (We have the issue for the `jpackage` failures, which are known, but we need to know if there is anything else that might show up.

**Comment by Haroon-Khel** [comment]
JDK 8 11 21 using the v1.0.1-release branch

test-azure-ubuntu2404-x64-1
        - [https://ci.adoptium.net/job/AQA_Test_Pipeline/282/console](https://ci.adoptium.net/job/AQA_Test_Pipeline/268/consolet)
test-docker-ubuntu2404-x64-1
        - [https://ci.adoptium.net/job/AQA_Test_Pipeline/283/console](https://ci.adoptium.net/job/AQA_Test_Pipeline/269/console)
test-docker-ubuntu2404-armv7-1
        - [https://ci.adoptium.net/job/AQA_Test_Pipeline/284/console](https://ci.adoptium.net/job/AQA_Test_Pipeline/270/console)
test-docker-ubuntu2404-armv8-1
        - [https://ci.adoptium.net/job/AQA_Test_Pipeline/285/console](https://ci.adoptium.net/job/AQA_Test_Pipeline/272/console)
test-docker-ubuntu2404-s390x-1
        - [https://ci.adoptium.net/job/AQA_Test_Pipeline/286/console](https://ci.adoptium.net/job/AQA_Test_Pipeline/273/console)
test-osuosl-ubuntu2404-aarch64-1
        - https://ci.adoptium.net/job/AQA_Test_Pipeline/287/console

**Comment by sxa** [comment]
Noting that we also have https://github.com/adoptium/infrastructure/issues/3598 to move from Ubuntu 23.10 (which will out of out support next month) on riscv64 to the LTS Ubuntu 24.04 release

**Comment by Haroon-Khel** [comment]
Next steps for this issue
>  Identifying older systems that can be replaced with Ubuntu 24.04 without restricting testing.

and

> Ideally having one real machine for each architecture on this

These 2 goals can be achieved together:
- test-azure-ubuntu2404-x64-1 and test-osuosl-ubuntu2404-aarch64-1 are already set up and in jenkins
- test-osuosl-ubuntu1604-ppc64le-1 and -2 could be upgraded to ubuntu2404
- We could commission a new marist ubuntu2404 machine, or upgrade our existing test-marist-ubuntu2004-s390x-1 to u2404
- Not sure about arm32, at the moment we do not have a vendor for actual arm32 machines

> Verifying that it is suitable as a base for our dockerhost system and deploying them: https://github.com/adoptium/infrastructure/issues/3585

https://ci.adoptium.net/computer/dockerhost-osuosl-ubuntu2404-aarch64-1/ has been in jenkins for a while, with one arm32 container running on it https://ci.adoptium.net/label/hw.dockerhost.arm.dockerhost-osuosl-ubuntu2404-aarch64-1/. We could try a combination of either upgrading existing/commissioning new dockerhost machines to ubuntu2404

> Add Ubuntu 24.04 to Vagrant Playbook Check job

This is blocked, see https://github.com/adoptium/infrastructure/issues/3577. A wider solution is to upgrade our existing VPC server to ubuntu2404, this is being discussed here https://github.com/adoptium/infrastructure/issues/3589

**Comment by Haroon-Khel** [comment]
> test-osuosl-ubuntu1604-ppc64le-1 and -2 could be upgraded to ubuntu2404

The ubuntu1604 osuosl ppc64le machine use a power8 cpu
```
Model name:                         POWER8 (architected), altivec supported
```
It looks like ubuntu2404 does not support power8 https://ubuntu.com/download/server/power

```
Starting with Ubuntu 22.04 LTS, POWER9 and POWER10 processors are supported.

The support for POWER8 ends with Ubuntu 21.10, respectively Ubuntu 20.04 LTS, which is still supported for several years.
```
During the command line upgrade:
```
Sorry, no more upgrades for this system 

There will not be any further Ubuntu releases for this system's 
POWER8 processor. 

Updates for Ubuntu 20.04 will continue until April 2025. 
```
It will have to be rebuilt in the osuosl console to a power 9 or 10 cpu. At the moment test-osuosl-ubuntu1604-ppc64le-1 has been upgraded to ubuntu2004

**Comment by sxa** [comment]
> It looks like ubuntu2404 does not support power8 https://ubuntu.com/download/server/power

Interesting - yeah we should definitely have some later machines in there. Do we definitely have access to provision POWER9/10 systems in our OSUOSL account? If so, let's go for it!

**Comment by Haroon-Khel** [comment]
Created https://ci.adoptium.net/computer/test-osuosl-ubuntu2404-ppc64le-1/, surprised it booted given my last attempts

```
root@test-osuosl-ubuntu2404-ppc64le-1:~# lscpu
Architecture:             ppc64le
  Byte Order:             Little Endian
CPU(s):                   4
  On-line CPU(s) list:    0-3
Model name:               POWER9 (architected), altivec supported
  Model:                  2.2 (pvr 004e 1202)
  Thread(s) per core:     1
  Core(s) per socket:     1
  Socket(s):              4
```
https://ci.adoptium.net/job/AQA_Test_Pipeline/349/console

**Comment by sxa** [comment]
Related - can we set up any "non-dockerhost" Ubuntu 24.04 test machines get labelled with [sw.tool.docker](https://ci.adoptium.net/label/sw.tool.docker) to indicate that test jobs which need to spin up containers can run on those hosts.

Related:
- https://github.com/adoptium/infrastructure/issues/1044
- https://github.com/adoptium/infrastructure/issues/3482
- https://github.com/adoptium/infrastructure/issues/2817

**Comment by Haroon-Khel** [comment]
Update on bare metal u2404 vms
x64: https://ci.adoptium.net/computer/test-azure-ubuntu2404-x64-1/
aarch64: https://ci.adoptium.net/computer/test-osuosl-ubuntu2404-aarch64-1/ 
s390x: https://ci.adoptium.net/computer/test-marist-ubuntu2404-s390x-1/
ppc64le: https://ci.adoptium.net/computer/test-osuosl-ubuntu2404-ppc64le-1/

We have a real machine for each of the architectures (the ones we have vendors for)

**Comment by sxa** [comment]
@Haroon-Khel Ref my last comment are all of those four "bare metal" systems you mentioned all able to run `docker`? In which case we should add the `sw.tool.docker` to them (It looks like only the s390x one has it so far).
Is this now closable with the machines now live and in active use?

**Comment by Haroon-Khel** [comment]
Ive updated the labels just now 👍🏻

**Comment by Haroon-Khel** [comment]
The only thing left on this epic is https://github.com/adoptium/infrastructure/issues/3577 which as @steelhead31 has commented is now unblocked

**Comment by sxa** [comment]
I'll leave it up to you but I'd be happy with closing this on the basis that it's good enough to be able to support and the VPC update is independent. But equally happy with leaving it for now if you'd prefer.

**Comment by Haroon-Khel** [comment]
Ok ill close this. The remaining tasks can be continued here https://github.com/adoptium/infrastructure/issues/3588
